### ISSUE TITLE
Define Anonymous Function via an attribute

### ISSUE THREAD

**Issue by willsmith9182** [bridgedotnet/Bridge#3890]
First off, I apologise if this is a bit badly explained. I've tried my best! 

I have 2 usages for defining an anonymous function, one looks like it could be solved by supporting a different module type, at least that's what this forum post is alluding to (https://forums.bridge.net/forum/community/help/6095-creating-a-node-js-module-with-exports-function-param -- more detail here)

Firstly the module.export pattern i require is not supported by any module types currently, but that's not what i'm asking in this issue (i'll probably raise another for the module.export stuff)
```js
module.exports = function(RED){

}
```

Next, inside this function i need to declare an anonymous function to act as a factory which is registered against the runtime to be invoked when a new instance of this step is required. What i'm really after is a way to define an entire method body, including params, and then have that method body compiled out to an anonymous JS function so that I can write definition classes for the objects going into the function. 

Currently I have hacked this about by declaring the function in a comment and then providing helper methods to get to the parameter being passed to this function but i was thinking, if i could define the function, and it's parameter(s) as a static on the class definition i would be able to get away from the helpers I've had to write. 

### Steps To Reproduce
this deck is my "actual" implementation with workarounds - I've cut a lot of the External types out
https://deck.net/458c86abe4c517e086706404256b50c9

```csharp
//pseudo-code as this attribute isn't supported!
// have also used dynamic for params, but in real life these are 2 external class definitions
public class MyCustomNode 
    {
        [Init(InitPosition.Bottom)]
        public static void MyCustomNodeInit()
        {
            /*@module.exports = function (RED) { */            
            // get the red runtime
            var runtime = Helpers.GetRuntime();
            
           MyCustomNode(runtime, null); // don't need to pass the config in here, but it would be cool if the method call was converted into an anonymous function using the template provided on the attribute which avoids me having to wrap the contents of the method in comments for form my method body.
            runtime.RegisterType("getdatasource", nameof(MyCustomNode));
            /*@} */
        }

       [AnonymousFunction("{@}({config})")] 
/* was thinking providing template like string for method body allows me to pass in references to things i need in the method without it being part of the method signature */
       public static void MyCustomNode(dynaimc runtime, dynamic config){
           var instance = Helpers.GetInstance<RedNodeInstance<MyCustomNode>>();
           // create the actual nodered instance

            runtime.CreateNodeInstance(instance, config);

	    /* call to instance.initialise (omitted for brevity) but it uses the config param!! */
       }
    }

```

### Expected Result

```js
const Bridge = require("./bridge.js");

/**
 * @compiler Bridge.NET 17.7.0
 */
Bridge.assembly("Demo", function ($asm, globals) {
    "use strict";

    Bridge.define("NodeRed.Base.MyCustomNode");
});

module.exports = function (RED) { 
var runtime = RED;

function MyCustomNode(config) { // this is the bit i want to generate without having to use comments
var instance = this;

runtime.nodes.createNode(instance, config);

/* call to instance.initialise (omitted for brevity) */

}
runtime.nodes.registerType("mycustomnode", MyCustomNode);
} 
```

### Actual Result

```js
// nope, because it's not supported :)
```

**Comment by geoffreymcgill** [comment]
This is a bit hard to follow. I'm not entirely sure I understand your requirements. Do you want to configure a C# Method to render as a closure?

Any clarity or further simplified code samples you can provide would be helpful. Or breaking this down into separate smaller issues.

**Comment by geoffreymcgill** [comment]
Closing for now. We will re-open if further details are provided.
